client/camera.lua

```diff
@@ -148,23 +144,20 @@ end)
 
 CreateThread(function()
 	while true do
-		if PlayerJob.name == "reporter" then
+		if QBX.PlayerData.job.name == "reporter" then
 			if holdingCam then
-				while not HasAnimDictLoaded(camanimDict) do
-					RequestAnimDict(camanimDict)
-					Wait(100)
-				end
+				lib.requestAnimDict(camanimDict)
 
-				if not IsEntityPlayingAnim(PlayerPedId(), camanimDict, camanimName, 3) then
-					TaskPlayAnim(GetPlayerPed(PlayerId()), 1.0, -1, -1, 50, 0, 0, 0, 0) -- 50 = 32 + 16 + 2
-					TaskPlayAnim(GetPlayerPed(PlayerId()), camanimDict, camanimName, 1.0, -1, -1, 50, 0, 0, 0, 0)
+				if not IsEntityPlayingAnim(cache.ped, camanimDict, camanimName, 3) then
+					TaskPlayAnim(GetPlayerPed(cache.playerId), 1.0, -1, -1, 50, 0, 0, 0, 0) -- 50 = 32 + 16 + 2
+					TaskPlayAnim(GetPlayerPed(cache.playerId), camanimDict, camanimName, 1.0, -1, -1, 50, 0, 0, 0, 0)
 				end
 
-				DisablePlayerFiring(PlayerId(), true)
+				DisablePlayerFiring(cache.playerId, true)
 				DisableControlAction(0,25,true) -- disable aim
 				DisableControlAction(0, 44,  true) -- INPUT_COVER
 				DisableControlAction(0,37,true) -- INPUT_SELECT_WEAPON
-				SetCurrentPedWeapon(PlayerPedId(), GetHashKey("WEAPON_UNARMED"), true)
+				SetCurrentPedWeapon(cache.ped, joaat("WEAPON_UNARMED"), true)
 				Wait(7)
```

Comment:
Convert to wait 0 as well, as waiting 7 will cause issues for the high end users